src/scheduler/fifo.c

```diff
@@ -520,6 +551,7 @@ init_scheduling_cycle(status *policy, int pbs_sd, server_info *sinfo)
 int
 schedule(int cmd, int sd, char *runjobid)
 {
+	update_svr_sched_state(SC_SCHEDULING);
```

Comment:
@arungrover / @bhroam 
As we all know scheduler can have one of the three states "down", "idle" and "scheduling".  Marking the scheduler state to either "down" or  "idle" from the server is straightforward because server knows when to kick off scheduling cycle.  To mark the state to "scheduling" is not trivial in the scenarios like the following.
* Let us say there are huge number of jobs in the system(like millions of jobs) then we really don't know how much time scheduler takes to complete scheduling them. During this time we should keep the state of scheduler as "scheduling" only.  This is possible only when scheduler updates to server that it has finished scheduling.  In the absence of this as per my knowledge I think server has no way to know that a scheduling cycle is over at so and so time. 
* Since most of the scheduler attributes can now be configured through qmgr during which an admin can make a mistake and scheduler might go down because of this.  In this case code changes are added to update server before scheduler actually goes down. This way there are less chances of race conditions.  But if we want to find this out from the server itself then it might take more time for it when compared to scheduler updating its own state to server.

Due to the above reasons I thought scheduler is the right person to update its state rather than somebody like server finding out the same.

Reply:
@suresh-thelkar The server has all the information it needs to update the scheduling attribute.  It did it before, and it still has it.  The server's state would update to scheduling during a scheduling cycle and turn back to Active when the cycle was over.  The server opens the connection to the scheduler.  At this point you update the state to scheduling.  When the scheduler closes the connection, the server updates the state to idle.

Reply:
@bhroam 
Thanks for your suggestion. I have made code changes accordingly.